data/tutorials/wf_03_metaprogramming.md

```diff
@@ -178,11 +182,16 @@ Let's look at a few properties of this tree:
 - A `structure_item` can either represent a toplevel expression, a type
   definition, a let definition, ... This is determined using a variant type.
 
-There are two complementary ways of getting a grasp on the Parsetree type. One
-is to read the API documentation, which includes examples of what each type and
-value represent. The other is to make OCaml dump the Parsetree value of crafted
-OCaml code. This is possible thanks to the option `-dparsetree`, available in
-the `ocaml` toplevel (as well as in `utop`):
+There are several complementary ways of getting a grasp on the Parsetree type.
+One is to read the [API
+documentation](https://v2.ocaml.org/api/compilerlibref/Parsetree.html)., which
+includes examples of what each type and value represent. Another is to make
+examine the Parsetree value of crafted OCaml code. This can be achieved using
+external tools such as [astexplorer](https://astexplorer.net/), the OCaml
+[VSCode
+extension](https://marketplace.visualstudio.com/items?itemName=ocamllabs.ocaml-platform),
+or even directly with the `ocaml` toplevel using the option `-dparsetree` (also
+available in `utop`):
```

Comment:
```suggestion
includes examples of what each type and value represent. Another is to 
examine the Parsetree value of crafted OCaml code. This can be achieved using
external tools such as [astexplorer](https://astexplorer.net/), the OCaml
[VSCode
extension](https://marketplace.visualstudio.com/items?itemName=ocamllabs.ocaml-platform),
or even directly with the OCaml toplevel using the option `-dparsetree` (also
available in UTop).
```